stm32-modules/include/thermocycler-refresh/thermocycler-refresh/tmc2130.hpp

```diff
@@ -0,0 +1,316 @@
+            return false;
+        }
+        if (!set_power_down_delay(
+                PowerDownDelay::reg_to_seconds(registers.tpowerdown.time),
+                policy)) {
+            return false;
+        }
+        if (!set_cool_threshold(registers.tcoolthrs, policy)) {
+            return false;
+        }
+        if (!set_thigh(registers.thigh, policy)) {
+            return false;
+        }
+        if (!set_chop_config(registers.chopconf, policy)) {
+            return false;
+        }
+        if (!set_cool_config(registers.coolconf, policy)) {
+            return false;
+        }
+        return true;
+    }
+
+    // FUNCTIONS TO SET/GET INDIVIDUAL REGISTERS
+
+    /**
+     * @brief Update GCONF register
+     * @param reg New configuration register to set
+     * @param policy Instance of abstraction policy to use
+     * @return True if new register was set succesfully, false otherwise
+     */
+    template <TMC2130Policy Policy>
+    auto set_gconf(GConfig reg, Policy& policy) -> bool {
+        // Assert that bits that MUST be 0 are actually 0
+        if ((reg.enc_commutation != 0) || (reg.test_mode != 0)) {
+            return false;
+        }
+        if (set_register(policy, reg)) {
+            _registers.gconfig = reg;
+            return true;
+        }
+        return false;
+    }
+    /**
+     * @brief Get the current GCONF register status. This register can
+     * be read, so this function gets it from the actual device.
+     */
+    template <TMC2130Policy Policy>
+    [[nodiscard]] auto get_gconf(Policy& policy) -> GConfig {
+        auto ret = read_register<GConfig>(policy);
+        if (ret.has_value()) {
+            _registers.gconfig = ret.value();
+        }
+        return _registers.gconfig;
+    }
+
+    /**
+     * @brief Get the general status register
+     */
+    template <TMC2130Policy Policy>
+    [[nodiscard]] auto get_gstatus(Policy& policy) -> GStatus {
+        auto ret = read_register<GStatus>(policy);
+        if (ret.has_value()) {
+            return ret.value();
+        }
+        return GStatus{.driver_error = 1};
+    }
+
+    /**
+     * @brief Update IHOLDIRUN register
+     * @param reg New configuration register to set
+     * @param policy Instance of abstraction policy to use
+     * @return True if new register was set succesfully, false otherwise
+     */
+    template <TMC2130Policy Policy>
+    auto set_current_control(CurrentControl reg, Policy& policy) -> bool {
+        // Assert that bits that MUST be 0 are actually 0
+        if ((reg.bit_padding_1 != 0) || (reg.bit_padding_2 != 0)) {
+            return false;
+        }
+        if (set_register(policy, reg)) {
+            _registers.ihold_irun = reg;
+            return true;
+        }
+        return false;
+    }
+    /**
+     * @brief Get the current IHOLDIRUN register status
+     */
+    [[nodiscard]] auto get_current_control() const -> CurrentControl {
```

Comment:
It's a little weird that the getters are inconsistent between retrieving values from the remote and returning the cached version. Especially since we have this whole other class for register map, IMO everything that's a getter in this class should be incurring a bus read and getting the remote status - even if it should be the same since the last write - and if you want cached values, you can check the map object.

Reply:
Issue is that a lot of these registers are write-only. GCONF is RW and CHOPCONF is RW (which I just realized), but everything else in the config map is Write-only

Reply:
then IMO we shouldn't have reads for those - just the register map read